streaming_jupyter_integrations/magics.py

```diff
@@ -474,6 +475,59 @@ def __create_configuration_from_dict(new_values: Dict[str, Any]) -> Configuratio
 
         return configuration
 
+    @line_magic
+    def flink_show_table_tree(self, line: str) -> Tree:
+        return self._build_schema_tree()
+
+    def _build_schema_tree(self) -> Tree:
+        tree = Tree()
+        catalogs = self.st_env.execute_sql("SHOW CATALOGS").collect()
+        for catalog in catalogs:
+            tree.add_node(self._build_catalog_node(catalog))
+        return tree
+
+    def _build_catalog_node(self, catalog) -> Node:
+        catalog_name = catalog[0]
+        self.st_env.execute_sql(f"USE CATALOG {catalog_name}")
```

Comment:
We are iterating over every catalog available and calling `USE CATALOG {catalog_name}` in every iteration. But we never go back to the originally set one. Is there any way to check which catalog is used, saving it in `_build_schema_tree` and then setting it back? Because right now, e.g., if I am using the default catalog and I call our magic, I'll end up using the last catalog on the list returned by `SHOW CATALOGS`.

Reply:
Yes, there is. In fact I had it implemented but it was not committed :D

Reply:
Yeah, I imagined that's why. I just wanted to make sure we "rollback" to the same state that we started with :)